app/validators/incomplete_postgraduate_course_details_validator.rb

```diff
@@ -0,0 +1,25 @@
+class IncompletePostgraduateCourseDetailsValidator < ActiveModel::EachValidator
+  include ActionView::Helpers::UrlHelper
+  include GovukLinkHelper
+  include GovukVisuallyHiddenHelper
+
+  def validate_each(record, attribute, application_choice)
+    return unless application_choice.application_form.teacher_degree_apprenticeship_feature_active?
+    return if application_choice.course.undergraduate?
```

Comment:
I put those two conditions in different lines to become easier to read and understand very fast. 
Let me know if you prefer everything in one liner.